agent/config_test.go

```diff
@@ -327,7 +327,11 @@ func TestDecodeConfig(t *testing.T) {
 		},
 		{
 			in: `{"http_api_response_headers":{"a":"b","c":"d"}}`,
-			c:  &Config{HTTPAPIResponseHeaders: map[string]string{"a": "b", "c": "d"}},
+			c:  &Config{DeprecatedHTTPAPIResponseHeaders: map[string]string{"a": "b", "c": "d"}},
```

Comment:
I think this should result only in the new field being set. The `Atlas` parameters are different since we no longer use them at all but there should be no code using the deprecated field.